app/controllers/event.js

```diff
@@ -0,0 +1,121 @@
+const Event = require('../models/Event')
+module.exports = {
+  createEvent: async (req, res, next) => {
+    const event = new Event(req.body)
+    try {
+      await event.save()
+      res.status(201).json({ event: event })
+    } catch (error) {
+      res.status(400).json({ error: error.errors })
+    }
+  },
+  updateEvent: async (req, res) => {
+    Event.findById(req.params.id, (err) => {
+      if (err) {
+        res.status(400).json({ error: 'Did not find an event!' })
+      } else {
+        Event.findByIdAndUpdate(req.params.id, req.body, function (err, updatedEvent) {
```

Comment:
in Event schema some of the fields have required set in it. So if a person have to change a particular field he have to enter all the fields again which he will not do. So i would suggest to check which fields are coming in req.body and append the fields which are not present in req.body by fetching them from db then save it

Reply:
sir for editing there will be a form, containing the previous values, user can simply edit the fields he/she likes. they won't be needing to fill whole 
and the new req.body will get replaced with the old set of values in database
.Hence , user will only alter  the values they like